src/AccordionToggle.js

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import AccordionContext from './AccordionContext';
+
+const propTypes = {
+  /**
+   * A key that corresponds to the collapse component that gets triggered
+   * when this has been clicked.
+   */
+  eventKey: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  as: 'div',
+};
+
+const AccordionToggle = ({ children, eventKey }) => {
+  const context = useContext(AccordionContext);
+
+  return (
+    <>
```

Comment:
then here do e.g.

```js
<Component onClick={...} (include other props) />
```

perhaps?

Reply:
so in the examples below then, you'd do `<Accordion.Toggle as="button" ...`

Reply:
So if I have `as` to have a default val of `button`, then the doc example would be updated to something like this?
```jsx
<Accordion defaultActiveKey="0">
  <Card>
    <Card.Header>
      <Accordion.Toggle
        onClick={key => console.log(`I've been clicked by : ${key}`)}
        eventKey="0"
      />
    </Card.Header>
    <Accordion.Collapse eventKey="0">
      <Card.Body>Hello! I'm the body</Card.Body>
    </Accordion.Collapse>
  </Card>
</Accordion>
```

Reply:
I think so, right? I'm looking at e.g. how `DropdownToggle` works. Then if you wanted to make the entire header clickable, you could do:

```js
<Accordion.Toggle
  as={Card.Header}
  // ...
>
```